pydantic/_internal/_generate_schema.py

```diff
@@ -1173,41 +1188,66 @@ def apply_model_validators(
 
 
 def apply_annotations(
```

Comment:
I'll note that replacing `apply_annotations` with this keeps all tests passing, reduces the size of the "fork" coming from `if metadata_get_schema is not None`, and eliminates the need for the interior binding of the metadata/metadata_get_schema that is only necessary due to being in the for loop:
```
def apply_annotations(
    get_inner_schema: ModifyCoreSchemaWrapHandler,
    annotations: typing.Iterable[Any],
    definitions: dict[str, core_schema.CoreSchema],
) -> ModifyCoreSchemaWrapHandler:
    """
    Apply arguments from `Annotated` or from `FieldInfo` to a schema.
    """
    for metadata in annotations:
        if metadata is None:
            continue
        get_inner_schema = get_wrapped_inner_schema(get_inner_schema, metadata, definitions)

    return get_inner_schema


def get_wrapped_inner_schema(
    get_inner_schema: ModifyCoreSchemaWrapHandler, metadata: Any, definitions: dict[str, core_schema.CoreSchema]
) -> Callable[[Any, ModifyCoreSchemaWrapHandler], core_schema.CoreSchema]:
    metadata_get_schema = getattr(metadata, '__get_pydantic_core_schema__', None) or (
        lambda source, handler: handler(source)
    )

    def _new_inner_schema_handler(
        source: Any, handler: ModifyCoreSchemaWrapHandler = get_inner_schema
    ) -> core_schema.CoreSchema:
        schema = metadata_get_schema(source, handler)
        schema = apply_single_annotation(schema, metadata, definitions)
        metadata_js_modify_function = _get_pydantic_modify_json_schema(metadata)
        CoreMetadataHandler(schema).compose_js_modify_functions(metadata_js_modify_function)
        return schema

    return _new_inner_schema_handler
```
Feel free to leave as is if you prefer

Reply:
Awesome, it is much simpler, implemented!